task_processing/plugins/kubernetes/task_config.py

```diff
@@ -113,16 +115,16 @@ def _valid_capabilities(capabilities: Sequence[str]) -> Tuple[bool, Optional[str
 
 
 class KubernetesTaskConfig(DefaultTaskConfigInterface):
-    def __invariant__(conf):
+    def __invariant__(self):
         return (
             (
-                len(conf.pod_name) < MAX_POD_NAME_LENGTH,
+                len(get_sanitised_kubernetes_name(self.pod_name)) < MAX_POD_NAME_LENGTH,
                 (
                     f'Pod name must have up to {MAX_POD_NAME_LENGTH} characters.'
                 )
             ),
             (
-                re.match(VALID_POD_NAME_REGEX, conf.pod_name),
+                re.match(VALID_POD_NAME_REGEX, get_sanitised_kubernetes_name(self.pod_name)),
```

Comment:
Sorry I didn't notice this earlier; should we be double-sanitising here, since `pod_name` is already `get_sanitised_kubernetes_name(self.name)`?